lib/statsd/instrument/datagram_builder.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+class StatsD::Instrument::DatagramBuilder
+  unless Regexp.method_defined?(:match?) # for ruby 2.3
+    module RubyBackports
+      refine Regexp do
+        def match?(str)
+          (self =~ str) != nil
+        end
+      end
+    end
+
+    using RubyBackports
+  end
+
+  def self.unsupported_datagram_types(*types)
+    types.each do |type|
+      define_method(type) do |_, _, _, _|
+        raise NotImplementedError, "Type #{type} metrics are not suppered by #{self.class.name}."
+      end
+    end
+  end
+
+  def initialize(prefix: nil, default_tags: nil)
+    @prefix = prefix.nil? ? "" : "#{normalize_name(prefix)}."
+    @default_tags = normalize_tags(default_tags)
+  end
+
+  def c(name, value, sample_rate, tags)
+    generate_generic_datagram(name, value, -'c', sample_rate, tags)
```

Comment:
I don't see the use for `String#-@` here. These are string literals inside a `frozen_string_literal: true` file, so they already are deduplicated.

Also it would break on MRI 2.3.